### PR TITLE
add an optional upload argument for an associated index ID

### PR DESCRIPTION
**Pull request by garobrik**

### CODE REVIEW

File: upload.go [hunk 1]
```diff
@@ -250,6 +254,10 @@ func (args requestArgs) EncodeQuery() string {
 	qs.SetOptionalInt("uploadId", args.uploadID)
 	qs.SetOptionalBool("done", args.done)
 
+	if args.associatedIndexID != nil {
+		qs.SetInt("associatedIndexID", *args.associatedIndexID)
```
Comment (efritz) [inline upload.go:258]: Just for consistency, do we have other lowerCamelCasedQueryParams that end in ID? If we do, is it `ID` or `Id`?

**Review by efritz — APPROVED** [review]

File: upload.go (hunk 1 above)
Comment (efritz) [reply upload.go:258]: https://github.com/sourcegraph/codeintelutils/blob/03d694caeb86d1399d2faab139a234b01c095956/upload.go#L254

Looks like `Id` is used just above.

Comment (garobrik) [reply upload.go:258]: woorps completely misread this comment, thought it was confirming ID was right lol

Comment (efritz) [reply upload.go:258]: 👅 I figured.

"thumbs up won't fix"

**Pull request merged**